src/LibraryManager/Providers/FileSystem/FileSystemProvider.cs

```diff
@@ -257,5 +195,12 @@ private static async Task<Stream> GetRemoteResourceAsync(string sourceUrl)
                 throw new ResourceDownloadException(sourceUrl);
             }
         }
+
+        protected override ILibraryNamingScheme LibraryNamingScheme { get; } = new SimpleLibraryNamingScheme();
+
+        protected override string GetDownloadUrl(ILibraryInstallationState state, string sourceFile)
+        {
+            throw new NotImplementedException();
```

Comment:
This is a little weird. No other way of doing it? I'm assuming this will never get called for the FileSystemProvider, still looks pretty screwy / bad smell. Maybe use "file:///" url, or re-design base class (create intermediate base class?)

Reply:
This isn't used by FileSystemProvider (it's only used in `RefreshCacheAsync` and FileSystemProvider doesn't use the cache), and doesn't really make sense to get a download URL in FS scenarios.  The other options here would be to either not make FileSystemProvider derive from BaseProvider, or - like you say - introduce an intermediate class for the InstallAsync implementation (which is the root of the code path where this comes into play).  Since FSProvider already overrides the InstallAsync implementation, this doesn't feel like a dirty architecture to me.

That said, the docs recommend using `NotSupportedException` in this case, so I'll change it to that.  Semantically, that's what I meant to say.